### PR TITLE
Milestone5 - Even handler and refactor code

### PR DESCRIPTION
**Pull request by janis-jenny**
- Add logic for numbers  in the handleClick listener on the App component
- Add logic for operations  in the handleClick listener on the App component
-  Refactor calculate method and operate 
-  Refactor render method on the App component

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,22 +1,112 @@
...
+        break;
+      case '=': {
+        const result = operate(total, next, operation);
+        const newState = { total: result, next: null, operation: null };
+        this.setState(newState);
+      }
+        break;
+      default:
+        break;
+    }
```
Comment (rukundoeric) [inline src/components/App.js]: - [x] <b>REQUIRED:</b> As it is stated in project specifications, all calculations operations should be done in calculate function from `logic/calculate.js` file. Kindly   Move this code to calculate function. The content of `handlClick` function should be something like this...
```Javascript
  handleClick = (buttonName) => {
     this.setState(calculate(this.state, buttonName));
  }
```

File: src/logic/operate.js [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
 import Big from 'big.js';
 
 const operate = (numberOne, numberTwo, operation) => {
   let result = Big(0);
   const a = Big(numberOne);
+  console.log(operation);
+  console.log('1', numberOne);
+  console.log('2', numberTwo);
   const b = Big(numberTwo);
+  console.log('b', b);
```
Comment (rukundoeric) [inline src/logic/operate.js]: - [x] <b>REQUIRED:</b> Make sure to always keep your code clean by making sure that you delete all debugging codes(console logs) when you no longer need them.

File: src/logic/operate.js [hunk 3]
```diff
@@ -1,9 +1,14 @@
+/* eslint-disable no-console */
```
Comment (rukundoeric) [inline src/logic/operate.js]: - [x] <b>REQUIRED:</b> This comment is no longer necessary here and besides, you are not allowed to use `eslint-disable` unless of course when it is totally obvious that there is no other way. kindly remove it.

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @janis-jenny , 

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 

## Required Changes ♻️
  - [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - N/A

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

File: src/logic/calculate.js [hunk 4]
```diff
@@ -22,29 +23,49 @@ const calculate = (data, btnName) => {
...
+        return { total: newTotal, next: newNext, operation: btnName };
       }
       break;
+    case '+':
+    case '-':
+    case 'X':
+    case '÷': {
+      const newOp = btnName;
+      return { total, next, operation: newOp };
+    }
```
Comment (rindrajosia) [inline src/logic/calculate.js]: - [ ] [REQUIRED] I cannot perform any operation with these operators. Please fix it.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/logic/calculate.js [hunk 5]
```diff
@@ -12,8 +11,10 @@ const calculate = (data, btnName) => {
 
   // Add a dot at the end of the recently typed number
   if (btnName === '.') {
+    if (!next && !total) return { total, next: `${0}${btnName}`, operation };
     if (!next) return { total: `${total}${btnName}`, next, operation };
     if (!total) return { total, next: `${next}${btnName}`, operation };
+    if (operation && total && (next !== '0' || next === '0')) return { total, next: `${next}${btnName}`, operation };
   }
```
Comment (rukundoeric) [inline src/logic/calculate.js]: - [ ] <b>OPTIONAL:</b> There is no need to use `if condition` and `switch case` separately in this function, you can simply move these conditions in the switch case below.

**Review by rukundoeric — APPROVED** [review]
Hi @janis-jenny,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - [ ] Kindly make sure to fix broken screenshots from your README before you merge this PR.


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
>As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**